src/connector/src/connector_common/common.rs

```diff
@@ -286,6 +289,13 @@ impl RdKafkaPropertiesCommon {
 
 impl KafkaCommon {
     pub(crate) fn set_security_properties(&self, config: &mut ClientConfig) {
+        // AWS_MSK_IAM
+        if self.is_aws_msk_iam() {
+            config.set("security.protocol", "SASL_SSL");
+            config.set("sasl.mechanism", "OAUTHBEARER");
```

Comment:
AWS_MSK_IAM is not a rdkafka value while OAUTHBEARER is. When using AWS_MSK_IAM, rdkafka mechanism will be set to OAUTHBEARER. I use AWS_MSK_IAM because we want't user to explicitly tell RW that an AWS service was used instead of implicitly by providing ak/sk.